changes/5243.added

```diff
@@ -0,0 +1 @@
+Added support for setting display_field on DynamicModelChoiceField to nested values in suggested choices list.
```

Comment:
We really need more visible documentation of the `display_field` property in general, but there's at least a little documentation in `nautobot/docs/development/jobs/index.md` at present that it would be good to update with this feature.

Reply:
Agreed - I initially looked for documentation, but dismissed that file since the feature did not seem relevant to jobs directly. I have now added few examples in the [commit](https://github.com/nautobot/nautobot/pull/5244/commits/e079989971b68b08131a8a5c4d603e336747103c) below.